packages/headless/src/controllers/knowledge/generated-answer/headless-answerapi-generated-answer.ts

```diff
@@ -211,3 +242,128 @@ function loadAnswerApiReducers(
   engine.addReducers({[answerApi.reducerPath]: answerApi.reducer, query});
   return true;
 }
+
+const buildAdvancedSearchQueryParams = (state: StateNeededByAnswerAPI) => {
+  const advancedSearchQueryParams = selectAdvancedSearchQueries(state);
+  const mergedCq = mergeAdvancedCQParams(state);
+
+  return {
+    ...advancedSearchQueryParams,
+    ...(mergedCq && {cq: mergedCq}),
+  };
+};
+
+const mergeAdvancedCQParams = (state: StateNeededByAnswerAPI) => {
```

Comment:
I believe we could use the following instead of create the function a second time: https://github.com/coveo/ui-kit/blob/4146434402013c70d353dcf59213feebf70839ba/packages/headless/src/features/search/search-request.ts#L190

Reply:
hmmm  they dont do the same thing though

- buildConstantQuery returns only the active tab's expression.
- mergeAdvancedCQParams returns a merged string of the active tab expression, static filter expressions, and advanced search cq, joined by " AND "

they are not equivalent as mergeAdvancedCQParams is a bit more comprehensive than the other one.

Reply:
Can you please be more precise on why they are different?

> buildConstantQuery returns only the active tab's expression.

I see `buildConstantQuery` returns the following: 
```
  return [cq, tabExpression, ...filterExpressions]
    .filter((expression) => !!expression)
    .join(' AND ');
```

Reply:
I thought it was combining multiple sources of filter expressions into a single query string, not just the active tab's expression. But after looking into `selectActiveTabExpression` its equivalent. You are right I could reuse it